### ISSUE TITLE
`regenerator-runtime/runtime` after `regenerator-runtime` does not work

### ISSUE THREAD

**Issue by julien-f** [facebook/regenerator#337]
`require('regenerator-runtime/runtime')` does not work if `regenerator-runtime` has been previously required.

```js
require('regenerator-runtime')

// regeneratorRuntime is undefined
console.log(global.regeneratorRuntime)

require('regenerator-runtime/runtime')

// regeneratorRuntime is still undefined
console.log(global.regeneratorRuntime)
```

[Repro repo](https://github.com/julien-f/regenerator-runtime-issue).

**Comment by julien-f** [comment]
I think we should clear the `require` cache at the end of `runtime-module.js`:

```js
delete require.cache[require('./path').path]
```

**Comment by benjamn** [comment]
Why not just define it globally (or better, just locally) yourself?

Doing anything with `require.cache` would make this code Node-specific, so that's not really an option.

**Comment by julien-f** [comment]
`babel-runtime@6` use `regenerator-runtime` but `@babel/preset-env@7` with the `polyfill` option is using `regenerator-runtime/runtime`.

So when I use a dependency using `@babel/preset-env` in my `babel-runtime` application, it fails with `regeneratorRuntime is not defined`.

**Comment by jquense** [comment]
lost a few hours trying to figure out why the runtime was inexplicably disappearing in a webpack app. this seems like really weird behavio (i'm sure there is a reason). Why not have the module version not add a global and have a polyfill import that requires it and appends to the global object which is a lot more common of a pattern for these sorts of things..

**Comment by benjamn** [comment]
Please note that #353 has landed in `regenerator-runtime@0.13.x`. While this version is not yet used by `@babel/runtime`, it changes the implementation of the `regenerator-runtime` package in ways that are likely to affect this issue (hopefully in a positive way!), so I would encourage you to review those changes if you have a minute.

**Comment by wmertens** [comment]
Aha! I'm using `pnpm` which uses a non-flat modules layout and the reason I'm getting a failing build is because I added `regenerator-runtime` with the wrong version to the top of the modules tree :-(
